Documentation/CodingGuidelines/CodingBestPractices/NamedArguments.rst

```diff
@@ -0,0 +1,247 @@
+..  include:: /Includes.rst.txt
+..  index:: pair: Coding guidelines; Named Arguments
+..  _cgl-named-arguments:
+
+===============
+Named Arguments
+===============
+
+`Named arguments <https://www.php.net/manual/en/functions.arguments.php#functions.named-arguments>`__,
+also known as “named parameters”, were `introduced in PHP 8 <https://wiki.php.net/rfc/named_params>`__,
+offering a new approach to passing arguments to functions. Instead of relying on
+the position of parameters, developers can now specify arguments based on their
+corresponding parameter names:
+
+..  code-block:: php
+    :caption: Named arguments example
+
+    <?php
+    function createUser($username, $email)
+    {
+        // code to create user
+    }
+    createUser(email: 'john.doe@example.com', username: 'john');
+
+This document discusses the use of named arguments within the TYPO3 core ecosystem,
+outlining best practices for TYPO3 extension and core developers regarding the
+adoption and avoidance of this language feature.
+
+
+Named Arguments in Public APIs
+==============================
+
+The key consideration when using this feature is outlined in the
+`PHP documentation <https://php.watch/versions/8.0/named-parameters>`__:
+
+    With named parameters, the name of the function/method parameters become part of
+    the public API, and changing the parameters of a function will be a semantic
+    versioning breaking-change. This is an undesired effect of named parameters feature.
+
+
+Utilizing Named Arguments in Extensions
+=======================================
+
+While the TYPO3 core cannot directly enforce or prohibit the use of named
+arguments within extensions, it suggests certain best practices to ensure
+forward compatibility:
+
+* Named arguments should only be used for initializing value objects
+  using PCPP (see below).
+
+* Avoid named arguments when calling TYPO3 core API methods unless dealing
+  with PCPP-based value objects. The TYPO3 core does not treat variable names as
+  part of the API and may change them without considering it a breaking change.
+
+
+TYPO3 Core Development
+======================
+
+The decision on when to employ named parameters within the TYPO3 core is carefully
+deliberated and codified into distinct sections, each subject to scrutiny within
+the Continuous Integration (CI) pipeline to ensure consistency and integrity over time.
+
+It’s important to note that the TYPO3 core team will not accept patches that aim
+to unilaterally transition the codebase from positional arguments to named arguments
+or vice versa without clear further benefits.
```

Comment:
patches just for the purpose of just adding named arguments for json_decode() everywhere will be denied. we want to avoid the review overhead when the patch does not come with true advantages but just because-we-can, and cgl masturbation.

Reply:
How about this addition to formulate it more welcoming and positive:

"Usage of named arguments following the recommendations is welcome for any TYPO3 Core patch that introduces new functionality." ?

Reply:
unsure. 'new functionality' could be understood as 'new feature'. 'clear further benefits' of a patch matches better. i don't think this sentence needs to be positive: we want to say which patches the core team will not accept.